src/ctapipe/image/muon/fitting.py

```diff
@@ -75,26 +112,36 @@ def taubin_circle_fit(x, y, mask):
     x_masked = x[mask]
     y_masked = y[mask]
 
-    R = x.max()  # x.max() just happens to be identical with R in many cases.
+    if weights is None:
+        weights_masked = np.ones(len(x_masked))
+    else:
+        weights_masked = weights[mask]
 
-    taubin_r_initial = R / 2
-    taubin_error = R * 0.1
-    xc = 0
-    yc = 0
+    r_initial = 1.1 * original_unit if r_initial is None else r_initial
```

Comment:
It does not make sense to use `1.1` in whatever unit was passed in. 1.1 only makes sense in degrees.